Change-Account Page, Settings Page, Meters Page

JustinRohweller opened the pull request:
These three pages are functional, yet need navigation, connection to back-end.

Commit:
First Commit

Background color change, text change

Commit:
Merged branch dev into Justin

Commit:
Changed index.android to use scenes folder

Commit:
Register file changed to overview

Commit:
Created Basic Overview Page

Graph page does nothing yet, plan to have graph page produce visuals which overview will pull and display at the bottom, overview has main menu button.

Commit:
DropDownMenu Added

first drop down menu added

Commit:
Merge branch 'dev'

Conflicts:
	index.android.js

Commit:
Settings Page Nearly Complete

I made a settings page that only needs to have its navigation connections set up and it is done.

Commit:
Change-Account Nearly Done

I made visuals for change-account page, although still needs to be modified, and have a function that connects to back-end to pick from different users. 
Started a meter page (just the file)

Commit:
Meters Page Added

Im about to mess up meters page by adding alerts within dropdown option selections

Commit:
Meter Page has Dropdown Alerts

^^

Commit:
Meters Page Updates

Meters (Add/Drop Meters) visuals and buttons fully made.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/dev'

Conflicts:
	index.android.js
	scenes/overview.js
	scenes/settings.js

Commit:
Merged Dev into Justin

Merged and added connections to change-account, meters, and settings pages

Commit:
Code Cleanup, Added button to meters page

Added device overview button to meters page.

Commit:
Code Cleanup, ready to merge

Committing prior to pull request with dev

jming422 commented on index.android.js:
```diff
@@ -42,7 +51,7 @@ const styles = StyleSheet.create({
     justifyContent: 'center',
     alignItems: 'center',
-    backgroundColor: '#F5FCFF',
+    backgroundColor: '#0000FF',
```
I think the background color was fine the way it was - I see no reason to change it

jming422 commented on scenes/change-account.js:
```diff
@@ -0,0 +1,167 @@
+//this file is the change account screen with a switch to account, account settings,
+//and deleting my account options.
+//I have no idea what account settings is supposed to be.
+//Still need to add registered users to this page, as well as who they are logged in as
```
These comments don't need to be in dev's version of this file

jming422 commented on scenes/change-account.js:
```diff
@@ -0,0 +1,167 @@
+              textStyle={styles.dropdown_text}
+              dropdownStyle={styles.dropdown_dropdown}
+              defaultValue='Switch to Which Account?'
+              renderRow={this.dropdown_renderRow.bind(this)}             
```
While this technically works, it's better practice for our app to bind scope to methods in the constructor rather than at call time.

jming422 commented on scenes/change-account.js:
```diff
@@ -0,0 +1,167 @@
+      <TouchableHighlight underlayColor='cornflowerblue'>
+        <View style={[styles.dropdown_row, {backgroundColor: evenRow ? 'lemonchiffon' : 'white'}]}>
+          <Text style={[styles.dropdown_row_text, highlighted && {color: 'mediumaquamarine'}]}>
+             {`${rowData}`}
```
This can be simplified to just `{rowData}`

jming422 commented on scenes/change-account.js:
```diff
@@ -0,0 +1,167 @@
+    borderWidth: 1,
+    borderRadius: 1,
+  },
+  dropdown_text: {
```
I'm pretty sure standard for these names is camelCase, not underscores

jming422 commented on scenes/settings.js:
```diff
@@ -1,21 +1,161 @@
+import ModalDropdown from 'react-native-modal-dropdown';
+
+//this file is the settings screen with options to logout, go to change account page, 
+//delete data history, or view contact info
```
These comments don't need to be in the dev version of this file

jming422 commented on scenes/settings.js:
```diff
@@ -1,21 +1,161 @@
+      <TouchableHighlight underlayColor='cornflowerblue'>
+        <View style={[styles.dropdown_row, {backgroundColor: evenRow ? 'lemonchiffon' : 'white'}]}>
+          <Text style={[styles.dropdown_row_text, highlighted && {color: 'mediumaquamarine'}]}>
+             {`${rowData}`}
```
Equivalent to `{rowData}`

jming422 commented on scenes/settings.js:
```diff
@@ -1,21 +1,161 @@
+    height: 1,
+    backgroundColor: 'cornflowerblue',
+  },
+});
```
Files should end with a newline

jming422 commented on scenes/change-account.js:
```diff
@@ -0,0 +1,167 @@
+    height: 1,
+    backgroundColor: 'cornflowerblue',
+  },
+});
```
Files should end with a newline

jming422 reviewed (changes requested):
New scenes are great! Lots of framework here that will be super useful going ahead. There's a good number of changes I've recommended, but they're all small things. Keep it up!

Commit:
Most Dev Push Changes Added

A little unsure how to make dropdowns into methods, so I am saving before I screw it up

jming422 commented on index.android.js:
```diff
@@ -20,7 +20,7 @@ export default class FlowApp extends Component {
     return (
       <Navigator
-        initialRoute={{ title: 'Flow', name: 'splash' }}
+        initialRoute={{ title: 'Flow', name: 'meters' }}
```
This file shouldn't be committed to dev, since the app needs to have the splash screen come up first. The rest of your changes are on point, though, good work so far! Nearly there it looks like.

jming422 reviewed (changes requested):
Nearly there! Making great progress 👍

Commit:
Fixed Dev Changes Completely

Ready to Merge Again

Commit:
Sorry Now Ready To Merge

Index points to splash, I just use pointing to meters and change-account and settings for testing

Commit:
I lied. Now it's ready to merge

Removed unnecessary imports.

Commit:
Styling Fixing-Don't Merge Until Monday

I am fixing my styling to match the splash and register page. Will be done by Monday

Commit:
Fixed Styling to Match Register and Login Page-Merge Ready

I believe it is fully ready to merge for real this time. Double Checked. It's a good thing we do these pull requests. Someone should look at it again still but I believe it is as good as it can get in terms of what we can do at the moment.

Commit:
Pointed Index.android to splash

Sorry fam. Should be good. I spam commits.

jming422 committed:
Final cleans, tweaks, and conflict resolves before merge

jming422 committed:
Resolve merge conflicts with dev

jming422 commented:
Woohoo! Let's get all this into dev 👍
